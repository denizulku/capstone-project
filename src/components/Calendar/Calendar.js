import React, { useState } from 'react'
import DayPicker from 'react-day-picker'
import List from '../List/List'
import 'react-day-picker/lib/style.css'
import styled from 'styled-components/macro'

const WEEKDAYS_SHORT = ['S', 'M', 'T', 'W', 'T', 'F', 'S']

export default function Calendar({ headline }) {
  const modifiers = {
    sundays: { daysOfWeek: [0] },
  }
  const [selectedDate, setSelectedDate] = useState(new Date().toDateString())
  const habits = JSON.parse(localStorage.getItem('habits'))

  const modifiersStyles = {
    sundays: {
      color: '#000000',
      backgroundColor: '#F0F0F0',
    },
  }

  const renderDay = (date) => {
    const habitsPerDay = habits
      .filter((habit) => {
        return (habit.completedDates || []).includes(date.toDateString())
      })
      .reduce((group, habit) => {
        if (habit.color in group) {
          group[habit.color]++
        } else {
          group[habit.color] = 1
        }

        return group
      }, {})

    return (
      <>
        <div>
          {date.getDate()}
          <div>
            {Object.entries(habitsPerDay).map(([color, count]) => {
              return (
                <div
                  style={{
                    backgroundColor: color,
                    borderRadius: '50%',
                  }}
                >
                  {count}
                </div>
              )
            })}
          </div>
        </div>
      </>
    )
  }
  return (
    <>
      <h2>{headline}</h2>
      <StyledCalendar>
        <DayPicker
          weekdaysShort={WEEKDAYS_SHORT}
          firstDayOfWeek={1}
          modifiers={modifiers}
          modifiersStyles={modifiersStyles}
          onDayClick={(day) => {
            setSelectedDate(day.toDateString())
          }}
          renderDay={renderDay}
        />
      </StyledCalendar>
      {selectedDate && (
        <List
          isReadOnly
          habits={habits.filter(
            (habit) => (habit.completedDates || []).indexOf(selectedDate) > -1
          )}
        />
      )}
    </>
  )
}

const StyledCalendar = styled.section`
  font-family: Roboto, sans-serif;
  display: flex;
  justify-content: center;
`
