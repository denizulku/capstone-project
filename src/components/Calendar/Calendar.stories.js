import React from 'react'
import Calendar from './Calendar'

export default {
  title: 'Calendar',
  component: Calendar,
}

const Template = (args) => <Calendar {...args} />

export const Primary = Template.bind({})
