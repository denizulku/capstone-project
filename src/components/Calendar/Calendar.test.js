import moment from 'moment'
import renderer from 'react-test-renderer'
import React from 'react'
import Calendar from './Calendar'

it('renders without crashing', () => {
  const props = {
    currentDay: moment('2017-12-26').format('MMM Do YYYY a'),
  }

  const tree = renderer.create(<Calendar {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
