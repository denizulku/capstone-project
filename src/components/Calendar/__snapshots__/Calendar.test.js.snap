// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="Calendar__StyledCalendar-sc-1e9ibe3-0 bWvkhB"
>
  <div
    className="DayPicker DayPicker--interactionDisabled"
    lang="en"
  >
    <div
      className="DayPicker-wrapper"
      onKeyDown={[Function]}
      tabIndex={0}
    >
      <div
        className="DayPicker-NavBar"
      >
        <span
          aria-label="Previous Month"
          className="DayPicker-NavButton DayPicker-NavButton--prev"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        />
        <span
          aria-label="Next Month"
          className="DayPicker-NavButton DayPicker-NavButton--next"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
          tabIndex="0"
        />
      </div>
      <div
        className="DayPicker-Months"
      >
        <div
          className="DayPicker-Month"
          role="grid"
        >
          <div
            className="DayPicker-Caption"
            role="heading"
          >
            <div
              onKeyUp={[Function]}
            >
              September 2020
            </div>
          </div>
          <div
            className="DayPicker-Weekdays"
            role="rowgroup"
          >
            <div
              className="DayPicker-WeekdaysRow"
              role="row"
            >
              <div
                className="DayPicker-Weekday"
                role="columnheader"
              >
                <abbr
                  title="Monday"
                >
                  M
                </abbr>
              </div>
              <div
                className="DayPicker-Weekday"
                role="columnheader"
              >
                <abbr
                  title="Tuesday"
                >
                  T
                </abbr>
              </div>
              <div
                className="DayPicker-Weekday"
                role="columnheader"
              >
                <abbr
                  title="Wednesday"
                >
                  W
                </abbr>
              </div>
              <div
                className="DayPicker-Weekday"
                role="columnheader"
              >
                <abbr
                  title="Thursday"
                >
                  T
                </abbr>
              </div>
              <div
                className="DayPicker-Weekday"
                role="columnheader"
              >
                <abbr
                  title="Friday"
                >
                  F
                </abbr>
              </div>
              <div
                className="DayPicker-Weekday"
                role="columnheader"
              >
                <abbr
                  title="Saturday"
                >
                  S
                </abbr>
              </div>
              <div
                className="DayPicker-Weekday"
                role="columnheader"
              >
                <abbr
                  title="Sunday"
                >
                  S
                </abbr>
              </div>
            </div>
          </div>
          <div
            className="DayPicker-Body"
            role="rowgroup"
          >
            <div
              className="DayPicker-Week"
              role="row"
            >
              <div
                aria-disabled={true}
                className="DayPicker-Day DayPicker-Day--outside"
              />
              <div
                aria-disabled={false}
                aria-label="Tue Sep 01 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={0}
              >
                1
              </div>
              <div
                aria-disabled={false}
                aria-label="Wed Sep 02 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                2
              </div>
              <div
                aria-disabled={false}
                aria-label="Thu Sep 03 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                3
              </div>
              <div
                aria-disabled={false}
                aria-label="Fri Sep 04 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                4
              </div>
              <div
                aria-disabled={false}
                aria-label="Sat Sep 05 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                5
              </div>
              <div
                aria-disabled={false}
                aria-label="Sun Sep 06 2020"
                aria-selected={false}
                className="DayPicker-Day DayPicker-Day--sundays"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                style={
                  Object {
                    "backgroundColor": "#F0F0F0",
                    "color": "#000000",
                  }
                }
                tabIndex={-1}
              >
                6
              </div>
            </div>
            <div
              className="DayPicker-Week"
              role="row"
            >
              <div
                aria-disabled={false}
                aria-label="Mon Sep 07 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                7
              </div>
              <div
                aria-disabled={false}
                aria-label="Tue Sep 08 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                8
              </div>
              <div
                aria-disabled={false}
                aria-label="Wed Sep 09 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                9
              </div>
              <div
                aria-disabled={false}
                aria-label="Thu Sep 10 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                10
              </div>
              <div
                aria-disabled={false}
                aria-label="Fri Sep 11 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                11
              </div>
              <div
                aria-disabled={false}
                aria-label="Sat Sep 12 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                12
              </div>
              <div
                aria-disabled={false}
                aria-label="Sun Sep 13 2020"
                aria-selected={false}
                className="DayPicker-Day DayPicker-Day--sundays"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                style={
                  Object {
                    "backgroundColor": "#F0F0F0",
                    "color": "#000000",
                  }
                }
                tabIndex={-1}
              >
                13
              </div>
            </div>
            <div
              className="DayPicker-Week"
              role="row"
            >
              <div
                aria-disabled={false}
                aria-label="Mon Sep 14 2020"
                aria-selected={false}
                className="DayPicker-Day DayPicker-Day--today"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                14
              </div>
              <div
                aria-disabled={false}
                aria-label="Tue Sep 15 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                15
              </div>
              <div
                aria-disabled={false}
                aria-label="Wed Sep 16 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                16
              </div>
              <div
                aria-disabled={false}
                aria-label="Thu Sep 17 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                17
              </div>
              <div
                aria-disabled={false}
                aria-label="Fri Sep 18 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                18
              </div>
              <div
                aria-disabled={false}
                aria-label="Sat Sep 19 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                19
              </div>
              <div
                aria-disabled={false}
                aria-label="Sun Sep 20 2020"
                aria-selected={false}
                className="DayPicker-Day DayPicker-Day--sundays"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                style={
                  Object {
                    "backgroundColor": "#F0F0F0",
                    "color": "#000000",
                  }
                }
                tabIndex={-1}
              >
                20
              </div>
            </div>
            <div
              className="DayPicker-Week"
              role="row"
            >
              <div
                aria-disabled={false}
                aria-label="Mon Sep 21 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                21
              </div>
              <div
                aria-disabled={false}
                aria-label="Tue Sep 22 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                22
              </div>
              <div
                aria-disabled={false}
                aria-label="Wed Sep 23 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                23
              </div>
              <div
                aria-disabled={false}
                aria-label="Thu Sep 24 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                24
              </div>
              <div
                aria-disabled={false}
                aria-label="Fri Sep 25 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                25
              </div>
              <div
                aria-disabled={false}
                aria-label="Sat Sep 26 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                26
              </div>
              <div
                aria-disabled={false}
                aria-label="Sun Sep 27 2020"
                aria-selected={false}
                className="DayPicker-Day DayPicker-Day--sundays"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                style={
                  Object {
                    "backgroundColor": "#F0F0F0",
                    "color": "#000000",
                  }
                }
                tabIndex={-1}
              >
                27
              </div>
            </div>
            <div
              className="DayPicker-Week"
              role="row"
            >
              <div
                aria-disabled={false}
                aria-label="Mon Sep 28 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                28
              </div>
              <div
                aria-disabled={false}
                aria-label="Tue Sep 29 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                29
              </div>
              <div
                aria-disabled={false}
                aria-label="Wed Sep 30 2020"
                aria-selected={false}
                className="DayPicker-Day"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="gridcell"
                tabIndex={-1}
              >
                30
              </div>
              <div
                aria-disabled={true}
                className="DayPicker-Day DayPicker-Day--outside"
              />
              <div
                aria-disabled={true}
                className="DayPicker-Day DayPicker-Day--outside"
              />
              <div
                aria-disabled={true}
                className="DayPicker-Day DayPicker-Day--outside"
              />
              <div
                aria-disabled={true}
                className="DayPicker-Day DayPicker-Day--sundays DayPicker-Day--outside"
                style={
                  Object {
                    "backgroundColor": "#F0F0F0",
                    "color": "#000000",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
